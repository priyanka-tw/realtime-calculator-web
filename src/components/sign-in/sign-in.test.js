import React from 'react';
import {act, fireEvent, render} from '@testing-library/react';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import SignIn from "./sign-in";
import Context from "../../services/global-context-provider/context";


describe('SignIn', () => {
    const mockedFn = jest.fn();
    it('should render the component', () => {

        const {container} = render(<SignIn />)

        expect(container).toMatchSnapshot();
    });

    it('should navigate trigger event on ws of login on click of start', async () => {
       const server = new WS("ws://localhost:8080/ws", { jsonProtocol: true });
        const client = new WebSocket("ws://localhost:8080/ws");

        const {getByTestId} = render(
            <Context.Provider value={{setUsername: mockedFn, socket: client}}>
            <SignIn />
            </Context.Provider>)

        await server.connected;
        await act(async () => {
            await fireEvent.change(getByTestId('username'), {target: {value: 'this-is-a-sample-username'}})
            await fireEvent.click(getByTestId('start-button'))
        })

        expect(mockedFn).toHaveBeenCalledWith('this-is-a-sample-username');
        await expect(server).toReceiveMessage({"data": "this-is-a-sample-username", "event": "login"});
    });
});
