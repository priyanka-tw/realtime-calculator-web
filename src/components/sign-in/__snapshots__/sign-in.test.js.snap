// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render the component 1`] = `
.c2 {
  color: white;
  width: 300px;
  font-size: 16pt;
  font-family: verdana;
  background-color: transparent;
  border: none;
  border-bottom: 2px solid #fff;
  outline: none;
  text-align: center;
  width: 400px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 400px;
  width: 500px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: black;
  margin-top: 100px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  margin-top: 20px;
  width: 200px;
  height: 30px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        class="c2"
        data-testid="username"
        placeholder="Please enter your name"
      />
      <button
        class="c3"
        data-testid="start-button"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
